python/cudf/cudf/testing/fuzzer.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+import functools
+import json
+import logging
+import os
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+
+try:
+    lru_cache = functools.lru_cache
+except Exception:
+    import functools32
+
+    lru_cache = functools32.lru_cache
+
+
+class Fuzzer(object):
+    def __init__(
+        self,
+        target,
+        data_handler_class,
+        dirs=None,
+        crash_reports_dir=None,
+        regression=False,
+        max_rows_size=4096,
+        max_cols_size=1000,
+        runs=-1,
+    ):
+
+        self._target = target
+        self._dirs = [] if dirs is None else dirs
+        self._crash_dir = crash_reports_dir
+        self._data_handler = data_handler_class(
+            dirs=self._dirs, max_rows=max_rows_size, max_columns=max_cols_size
+        )
+        self._total_executions = 0
+        self._regression = regression
+        self._start_time = None
+        self.runs = runs
+
+    def log_stats(self):
+        end_time = datetime.datetime.now()
+        total_time_taken = end_time - self._start_time
+
+        logging.info(f"Run-Time elapsed (hh:mm:ss.ms) {total_time_taken}")
+
+    def write_crash(self, error):
+        error_file_name = datetime.datetime.now().__str__()
+        if self._crash_dir:
+            crash_path = os.path.join(
+                self._crash_dir, error_file_name + "_crash.json",
+            )
+            crash_log_path = os.path.join(
+                self._crash_dir, error_file_name + "_crash.log",
+            )
+        else:
+            crash_path = error_file_name + "_crash.json"
+            crash_log_path = error_file_name + "_crash.log"
+
+        with open(crash_path, "w") as f:
+            json.dump(
+                self._data_handler.current_params, f, sort_keys=True, indent=4
+            )
+
+        logging.info(f"Crash params was written to {crash_path}")
+
+        with open(crash_log_path, "w") as f:
+            f.write(str(error))
+        logging.info(f"Crash exception was written to {crash_log_path}")
+
+    def start(self):
+
+        while True:
+            logging.info(f"Running test {self._total_executions}")
+            file_name = self._data_handler.generate_input()
+            try:
+                self._start_time = datetime.datetime.now()
+                self._target(file_name)
+            except Exception as e:
```

Comment:
We should be more specific about the types of exceptions we handle here. If someone sends a `ctrl + C` to kill the loop we shouldn't catch it here.

Reply:
Handled Keyboard interrupts separately and breaking out of the loop. Else, we'd want to log the failure and capture crash params.